bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/handler/ZwaveJSBridgeHandler.java

```diff
@@ -182,6 +182,17 @@ public void onEvent(BaseMessage message) {
                         nodeListener.onNodeDead(eventMsg.event);
                     }
                     break;
+                case "node removed":
+                    if (nodeListener != null) {
+                        nodeListener.onNodeRemoved(eventMsg.event);
+                    }
+                    break;
+                case "node added":
+                    final NodeDiscoveryService discovery = discoveryService;
+                    if (discovery != null) {
+                        discovery.addNodeDiscovery(eventMsg.event.node);
+                    }
+                    break;
```

Comment:
This part is the actual fix, as these cases where not routed to responsible parts of the binding.